### PR TITLE
chore: Remove deprecations for user lib

### LINKED ISSUE #11607: Remove deprecations for user lib
Based on the refactor of the user related components, we need to clean up the deprecations with the first next major release. 

- remove the login page template styles from the styling lib
- remove forgot email password module
- remove update email module
- remove transitional tokes and transitional modules (will be possible when all depending features will be moved out from `@spartacus/core`)
- remove redundant translations from the lib `@spartacus/assets` (otherwise there's an i18n warning - see #11926)

### PR DESCRIPTION
**Pull request by Matejk00**
Closes: #11607

### CODE REVIEW

File: projects/assets/src/translations/en/common.ts [hunk 1]
```diff
@@ -120,11 +120,6 @@ export const common = {
     total: '{{total}}',
     count: '{{count}}',
   },
-  // deprecated since 3.2, moved to @spartacus/user/account/assets
-  miniLogin: {
-    userGreeting: 'Hi, {{name}}',
-    signInRegister: 'Sign In / Register',
-  },
```
Comment (marlass) [inline projects/assets/src/translations/en/common.ts:127]: List all translation keys that were moved to other libs in the migration doc.

**Review by marlass — CHANGES_REQUESTED** [review]
Everything that was removed from public API should have migration with `removed-public-api-deprecation` mechanism with appropiate comments where users can find new version (components/modules/services/constants).

Components/services that were removed from public api should also be listed in the doc (not only modules).

Also mention that styles for user components were removed from styles lib

File: projects/assets/src/translations/en/common.ts (hunk 1 above)
Comment (Matejk00) [reply projects/assets/src/translations/en/common.ts:127]: done

File: docs/migration/4_0.md [hunk 2]
```diff
@@ -1,3 +1,73 @@
...
+- Key `register.password.label` was moved to separated `@spartacus/user` lib.
+- Key `register.password.placeholder` was moved to separated `@spartacus/user` lib.
+- Key `register.newPassword` was moved to separated `@spartacus/user` lib.
+- Key `register.emailMarketing` was moved to separated `@spartacus/user` lib.
+- Key `register.confirmThatRead` was moved to separated `@spartacus/user` lib.
+- Key `register.selectTitle` was moved to separated `@spartacus/user` lib.
+- Key `register.passwordMinRequirements` was moved to separated `@spartacus/user` lib.
+- Key `register.bothPasswordMustMatch` was moved to separated `@spartacus/user` lib.
+- Key `register.titleRequired` was moved to separated `@spartacus/user` lib.
+- Key `register.postRegisterMessage` was moved to separated `@spartacus/user` lib.
```
Comment (marlass) [inline docs/migration/4_0.md]: Would be good to mention for all of these keys that they are in `userAccountTranslations`, `userAccountTranslationChunksConfig` from `@spartacus/user/account/assets` and in `userProfileTranslationChunksConfig`, `userProfileTranslations` from `@spartacus/user/profile/assets`

File: projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts [hunk 3]
```diff
@@ -34,6 +61,97 @@ export const REMOVED_PUBLIC_API_DATA: DeprecatedNode[] = [
...
+  {
+    node: CLOSE_ACCOUNT_COMPONENT,
+    importPath: SPARTACUS_STOREFRONTLIB,
+    comment: `'${CLOSE_ACCOUNT_COMPONENT}' was moved to ${SPARTACUS_USER_PROFILE_COMPONENTS}.`,
+  },
+  // projects/storefrontlib/src/cms-components/myaccount/close-account/components/close-account-modal/close-account-modal.component.ts
+  {
+    node: CLOSE_ACCOUNT_MODAL_COMPONENT,
+    importPath: SPARTACUS_STOREFRONTLIB,
+    comment: `'${CLOSE_ACCOUNT_MODAL_COMPONENT}' was moved to ${SPARTACUS_USER_PROFILE_COMPONENTS}.`,
```
Comment (marlass) [inline projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts]: Can we mention in here and the docs, that there are slight changes between these two components? (mostly constructor)?

File: docs/migration/4_0.md [hunk 4]
```diff
@@ -1,3 +1,73 @@
 # Technical Changes in Spartacus 4.0
 
 ## Breaking Changes Introduced in 4.0
+### CMS Components
+
+- Following modules `CloseAccountModule`, `ForgotPasswordModule`, `UpdateEmailModule`, `UpdatePasswordModule`, `UpdateProfileModule` were moved to separated `@spartacus/user` lib.
+- Component `UpdateEmailFormComponent` was removed. For replacement `UpdateEmailComponent` from `@spartacus/user` can be used.
+- Component `UpdateProfileFormComponent` was removed. For replacement `UpdateProfileComponent` from `@spartacus/user` can be used.
+- Component `UpdatePasswordFormComponent` was removed. For replacement `UpdatePasswordComponent` from `@spartacus/user` can be used.
+- Componenets `CloseAccountComponent`, `CloseAccountModalComponent`, `ForgotPasswordComponent`, `UpdateEmailComponent`, `UpdatePasswordComponent`, `UpdateProfileComponent` were moved to separated `@spartacus/user` lib.
```
Comment (marlass) [inline docs/migration/4_0.md]: Can we point to more precise entry points where we can find these components/modules? Eg. `@spartacus/user/profile/components`

File: projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts [hunk 5]
```diff
@@ -34,6 +61,97 @@ export const REMOVED_PUBLIC_API_DATA: DeprecatedNode[] = [
...
+  {
+    node: FORGOT_PASSWORD_MODULE,
+    importPath: SPARTACUS_STOREFRONTLIB,
+    comment: `'${FORGOT_PASSWORD_MODULE}' was moved to ${SPARTACUS_USER_PROFILE_COMPONENTS}.`,
+  },
+  // projects/storefrontlib/src/cms-components/myaccount/forgot-password/forgot-password.component.ts
+  {
+    node: FORGOT_PASSWORD_COMPONENT,
+    importPath: SPARTACUS_STOREFRONTLIB,
+    comment: `'${FORGOT_PASSWORD_COMPONENT}' was moved to ${SPARTACUS_USER_PROFILE_COMPONENTS}.`,
```
Comment (marlass) [inline projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts]: Can we mention about the changes to the component here and in docs (changed html, separated component service, changes in the properties/methods)?

File: projects/storefrontlib/src/cms-components/myaccount/index.ts [hunk 6]
```diff
@@ -1,16 +1,10 @@
 export * from './address-book/index';
-export * from './close-account/index';
 export * from './consent-management/index';
-export * from './forgot-password/forgot-password.component';
-export * from './forgot-password/forgot-password.module';
 export * from './order/index';
 export * from './payment-methods/payment-methods.component';
 export * from './payment-methods/payment-methods.module';
 export * from './reset-password/reset-password-form.component';
 export * from './reset-password/reset-password.module';
```
Comment (marlass) [inline projects/storefrontlib/src/cms-components/myaccount/index.ts]: Shouldn't also these two things be removed and mentioned in this PR?

File: projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts [hunk 7]
```diff
@@ -34,6 +61,97 @@ export const REMOVED_PUBLIC_API_DATA: DeprecatedNode[] = [
...
+    node: UPDATE_EMAIL_FORM_COMPONENT,
+    importPath: SPARTACUS_STOREFRONTLIB,
+    comment: `'${UPDATE_EMAIL_FORM_COMPONENT}' was removed. For replacement use '${UPDATE_EMAIL_COMPONENT}' from ${SPARTACUS_USER}.`,
+  },
+  // projects/storefrontlib/src/cms-components/myaccount/update-email/update-email-form/update-email-form.component.ts
+  {
+    node: UPDATE_EMAIL_COMPONENT,
+    importPath: SPARTACUS_STOREFRONTLIB,
+    comment: `'${UPDATE_EMAIL_COMPONENT}' was moved to ${SPARTACUS_USER_PROFILE_COMPONENTS}.`,
+  },
```
Comment (marlass) [inline projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts]: For these 2 can you provide more info in docs how it changed? New service etc.

File: projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts [hunk 8]
```diff
@@ -34,6 +61,97 @@ export const REMOVED_PUBLIC_API_DATA: DeprecatedNode[] = [
...
+    node: UPDATE_PASSWORD_FORM_COMPONENT,
+    importPath: SPARTACUS_STOREFRONTLIB,
+    comment: `'${UPDATE_PASSWORD_FORM_COMPONENT}' was removed. For replacement use '${UPDATE_PASSWORD_COMPONENT}' from ${SPARTACUS_USER}.`,
+  },
+  // projects/storefrontlib/src/cms-components/myaccount/update-password/components/update-password/update-password.component.ts
+  {
+    node: UPDATE_PASSWORD_COMPONENT,
+    importPath: SPARTACUS_STOREFRONTLIB,
+    comment: `'${UPDATE_PASSWORD_COMPONENT}' was moved to ${SPARTACUS_USER_PROFILE_COMPONENTS}.`,
+  },
```
Comment (marlass) [inline projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts]: Same story wiith these. What changed, new architecture, combined components, new service

File: projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts [hunk 9]
```diff
@@ -34,6 +61,97 @@ export const REMOVED_PUBLIC_API_DATA: DeprecatedNode[] = [
...
+    node: UPDATE_PROFILE_MODULE,
+    importPath: SPARTACUS_STOREFRONTLIB,
+    comment: `'${UPDATE_PROFILE_MODULE}' was moved to ${SPARTACUS_USER_PROFILE_COMPONENTS}.`,
+  },
+  // projects/storefrontlib/src/cms-components/myaccount/update-profile/update-profile.component.ts
+  {
+    node: UPDATE_PROFILE_COMPONENT,
+    importPath: SPARTACUS_STOREFRONTLIB,
+    comment: `'${UPDATE_PROFILE_COMPONENT}' was moved to ${SPARTACUS_USER_PROFILE_COMPONENTS}.`,
+  },
```
Comment (marlass) [inline projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts]: Sam with these. Also the UpdateProfileFormComponent config is missing here

File: docs/migration/4_0.md [hunk 10]
```diff
@@ -1,3 +1,73 @@
...
+- Key `register.confirmThatRead` was moved to separated `@spartacus/user` lib.
+- Key `register.selectTitle` was moved to separated `@spartacus/user` lib.
+- Key `register.passwordMinRequirements` was moved to separated `@spartacus/user` lib.
+- Key `register.bothPasswordMustMatch` was moved to separated `@spartacus/user` lib.
+- Key `register.titleRequired` was moved to separated `@spartacus/user` lib.
+- Key `register.postRegisterMessage` was moved to separated `@spartacus/user` lib.
+
+### Changes in styles
+
+- All styles releated to user components (under `scss/componenets/user` folder) were moved to  `@spartacus/user` lib.
```
Comment (marlass) [inline docs/migration/4_0.md]:
```suggestion
- All styles releated to user components (under `scss/components/user` folder) were moved to  `@spartacus/user` lib.
```

File: projects/storefrontstyles/scss/components/index.scss [hunk 11]
```diff
@@ -3,7 +3,6 @@
 @import './content/index';
 @import './product/index';
 @import './misc/index';
-@import './user/index';
```
Comment (marlass) [inline projects/storefrontstyles/scss/components/index.scss:6]: Regarding styles I think some things were forgotten when creating user lib.
eg. styles for close-account were not moved.

We should also remove some of them from `my-account` directory in storefrontstyles.

Could you evaluate what was forgotten and move it to user lib and then removed the duplicates from storefrontstyles?

**Review by marlass — CHANGES_REQUESTED** [review]

File: docs/migration/4_0.md [hunk 12]
```diff
@@ -1,3 +1,156 @@
...
+- Component `UpdateProfileFormComponent` was removed. For replacement `UpdateProfileComponent` from `@spartacus/user/profile/components` can be used.
+- Components `CloseAccountComponent`, `CloseAccountModalComponent`, `ForgotPasswordComponent`, `RegisterComponent`, `UpdateEmailComponent`, `UpdatePasswordComponent`, `UpdateProfileComponent` were moved to `@spartacus/user/profile/components`. Logic for those components was changed. For details look on sections below.
+- Components `LoginComponent`, `LoginFormComponent` were moved to `@spartacus/user/account/components`. Logic for those components was changed. For details look on sections below.
+- Component `LoginRegisterComponent` were moved to `@spartacus/user/account/components`.
+
+#### CloseAccountModalComponent
+
+- All services used in constructor have been changed to be `protected`.
+- Component is not using `UserService` anymore, `UserProfileFacade` was introduced.
+- Component is no longer using `Subscription`, also `ngOnDestroy` method was removed.
```
Comment (marlass) [inline docs/migration/4_0.md]:
```suggestion
- Component is no longer using `subscription` property, also `ngOnDestroy` method was removed.
```

File: docs/migration/4_0.md [hunk 13]
```diff
@@ -1,3 +1,156 @@
 # Technical Changes in Spartacus 4.0
 
 ## Breaking Changes Introduced in 4.0
+
+### CMS Components
+
+- Following modules `CloseAccountModule`, `ForgotPasswordModule`, `RegisterModule`, `ResetPasswordModule`, `UpdateEmailModule`, `UpdatePasswordModule`, `UpdateProfileModule` were moved to `@spartacus/user/profile/components`.
```
Comment (marlass) [inline docs/migration/4_0.md]:
```suggestion
- Following modules `CloseAccountModule`, `ForgotPasswordModule`, `RegisterComponentModule`, `ResetPasswordModule`, `UpdateEmailModule`, `UpdatePasswordModule`, `UpdateProfileModule` were moved to `@spartacus/user/profile/components`.
```

File: docs/migration/4_0.md [hunk 14]
```diff
@@ -1,3 +1,156 @@
...
+- Services `RoutingService`, `UserService`, `GlobalMessageService`, are no longer used directly in component file. New service `UpdatePasswordComponentService` was introduced and used in constructor.
+- Logic for `onSubmit` method was changed. Now this method has no parameters.
+- New properties `form`, `isUpdating` were added.
+- There were important change in component template. Since `UpdatePasswordFormComponent` was removed `UpdatePasswordComponent` contains now the template for update password form itself.
+- Change detection strategy for this component was set to `OnPush`.
+- Spinner component was added to the template which relys on `isUpdating` property.
+
+#### UpdateProfileComponent
+
+- Properties `user`, `loading` were removed.
```
Comment (marlass) [inline docs/migration/4_0.md]:
```suggestion
- Properties `user$`, `loading$` were removed.
```

File: docs/migration/4_0.md [hunk 15]
```diff
@@ -1,3 +1,156 @@
...
+- Methods `ngOnInit`, `onCancel`, `onSuccess`, `ngOnDestroy` were removed from component file.
+- Services `RoutingService`, `UserService`, `GlobalMessageService`, are no longer used directly in component file. New service `UpdateProfileComponentService` was introduced and used in constructor.
+- Logic for `onSubmit` method was changed. Now this method has no parameters.
+- New properties `form`, `isUpdating` were added.
```
Comment (marlass) [inline docs/migration/4_0.md]:
```suggestion
- New properties `form`, `isUpdating$` were added.
```

File: docs/migration/4_0.md [hunk 16]
```diff
@@ -1,3 +1,156 @@
...
+
+#### LoginComponent
+
+- Type of `user` property was changed to `Observable<User | undefined>` instead of `Observable<User>`.
+
+#### LoginFormComponent
+
+- Services `AuthService`, `GlobalMessageService`, `FormBuilder`, `WindowRef` are no longer used directly in component file. New service `LoginFormComponentService` was introduced and used in constructor.
+- Methods `ngOnInit`, `submitForm`, `loginUser` were removed from component file.
+- New properties `form`, `isUpdating` were added.
```
Comment (marlass) [inline docs/migration/4_0.md]:
```suggestion
- New properties `form`, `isUpdating$` were added.
```

File: docs/migration/4_0.md [hunk 17]
```diff
@@ -1,3 +1,156 @@
...
+- Type of `token` property was changed from `string` to `Observable<string>`.
+- Property `resetPasswordForm` was renamed to `form`.
+- New observable property `isUpdating` was added.
+- Methods `ngOnInit`, `resetPassword`, `ngOnDestroy` were removed. New method `onSubmit` was added.
+- Services `FormBuilder`, `UserService`, `RoutingService`, are no longer used directly in component file. New service `ResetPasswordComponentService` was introduced and used in constructor.
+- Change detection strategy for this component was set to `OnPush`.
+- Component template was adapted to the new logic. 
+- Spinner component was added which relys on `isUpdating` property.
+
+#### LoginComponent
```
Comment (marlass) [inline docs/migration/4_0.md:50]: Also worth to mention changes in constructor of this component

File: docs/migration/4_0.md [hunk 18]
```diff
@@ -1,3 +1,156 @@
...
+#### ForgotPasswordComponent 
+
+- Property `forgotPasswordForm` was renamed to `form`.
+- New observable property `isUpdating` was added.
+- Methods `ngOnInit`, `requestForgotPasswordEmail` were removed. New method `onSubmit` was added.
+- Services `FormBuilder`, `UserService`, `RoutingService`, `AuthConfigService` are no longer used directly in component file. New service `ForgotPasswordComponentService` was introduced and used in constructor.
+- Change detection strategy for this component was set to `OnPush`.
+- There were slight changes in component template. Spinner component was added which relys on `isUpdating` property, also form now is using `onSubmit` method on form submit event.
+
+#### RegisterComponent
```
Comment (marlass) [inline docs/migration/4_0.md:32]: Also mention changes in constructor

File: feature-libs/user/profile/styles/_index.scss [hunk 19]
```diff
@@ -1 +1,4 @@
+@import './close-account';
+@import './close-account-modal';
 @import './forget-password';
+@import './update-password-form';
```
Comment (marlass) [inline feature-libs/user/profile/styles/_index.scss:5]: I think we should also move register scss from account to profile

**Review by marlass — COMMENTED** [review]
I also wonder why we have _user-form.scss file in the profile styles

**Review by marlass — APPROVED** [review]

**Comment by marlass** [comment]
Merged in `epic/4-0-deprecations`

**Pull request closed without merging**